### PR TITLE
refactor: rewrite entire library to use axios and typescript

### PR DESCRIPTION
**Pull request by jpoehnelt**
This simplifies the library significantly and exposes the entire inner workings of [axios](https://github.com/axios/axios) to the user and inlines [@types/google__maps](https://www.npmjs.com/package/@types/google__maps).

### Definitely
- [x] implement all methods(removed speedlimits method for now)
- [x] implement unit tests
- [x] implement e2e tests
- [x] removed ratelimiting/throttling code that was failing
- [x] support keepalive connections and connection pooling
- [x] rename to `@googlemaps/google-maps-services-node` or `@googlemaps/node`. other updates to package.json as necessary
- [x] fix package and test
- [x] client pattern
- [x] experience id #301 
### Probably
- [x] ~incorporate polyline encoding~
- [x] usage example in README.md
- [ ] typedoc
### Maybe
- [ ] remove place types and change to string instead of enum
- [ ] add support for premium accounts
- [ ] retries
- [ ] logging
- [x] standardize `key` in params interface
- [x] convert polyline encoding/decoding to typescript

### CODE REVIEW

File: e2e/geocode/geocode.test.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    components: { country: "us" },
+    bounds: {
+      northeast: { lat: 50, lng: -110 },
+      southwest: { lat: 35, lng: -130 }
+    },
+    key: process.env.GOOGLE_MAPS_API_KEY
+  };
+
+  const r = await geocode({ params: params });
+  expect(r.data.results[0].place_id).toEqual("ChIJVTPokywQkFQRmtVEaUZlJRA");
```
[the remark this answers is not in the document]
Comment (jpoehnelt) [reply e2e/geocode/geocode.test.ts]: all the tests under e2e are hitting production services. i tried to keep the assertion fairly generic. i will make another pass

File: package.json [hunk 2]
```diff
@@ -1,28 +1,18 @@
 {
-  "name": "@google/maps",
-  "version": "1.0.2",
+  "name": "@googlemaps/google-maps-services-node",
+  "version": "2.0.0-rc.1",
```
[the remark this answers is not in the document]
Comment (jpoehnelt) [reply package.json]: I think i'm going to keep it in a non master branch for a bit and publish the rc from there. I'm waiting on https://github.com/sindresorhus/query-string/pull/234 as I prefer to not use a fork on github.

[the remark this answers is not in the document]
Comment (jpoehnelt) [reply package.json]: Reverted this change.

[the remark this answers is not in the document]
Comment (arriolac) [reply package.json]: > I think i'm going to keep it in a non master branch for a bit and publish the rc from there. I'm waiting on sindresorhus/query-string#234 as I prefer to not use a fork on github.

Sounds good. Did you want to keep in this branch? Or merge into a version branch? (e.g. `v2.0.0`)

[the remark this answers is not in the document]
Comment (jpoehnelt) [reply package.json]: version branch sounds better

**Pull request merged**